src/main/scala/NIC.scala

```diff
@@ -448,12 +491,12 @@ trait CanHavePeripheryIceNICModuleImp extends LazyModuleImp {
 
     if (nicConf.usePauser) {
       val pauser = Module(new PauserComplex(qDepth))
-      pauser.io.ext.flipConnect(NetDelay(netio, latency))
+      pauser.io.ext.flipConnect(NetDelay(NICIO(netio), latency))
       pauser.io.int.out <> pauser.io.int.in
       pauser.io.macAddr := netio.macAddr + (1 << 40).U
       pauser.io.settings := netio.pauser
     } else {
-      netio.in <> Queue(LatencyPipe(netio.out, latency), qDepth)
+      netio.in <> Queue(LatencyPipe(NICIO(netio).out, latency), qDepth)
```

Comment:
Have you tested that this case actually elaborates correctly? I'm pretty sure that, since `netio.in` no longer has the ready signal, this will cause chisel to complain about an unconnected wire. What you should do is assign `NICIO(netio)` to a wire and then connect that signals in and out to each other through a pipe.

Reply:
If I understand this correctly, `qDepth` is meaningless here then? Thus the code should be 
`      netio.in := Pipe(NICIO(netio).out, latency)`

Reply:
The above seems to work (nic-loopback succeeds)